File: CX/ClientApp/src/servicios/ServicioTipoInteraccion.js
import { ProcesarDatosApi } from "./ApiFetch";
const controlador = "tipointeraccion";

export const AgregarTipoInteraccion = async (data) => {
    const url = `${controlador}/insertartipointeraccion`;
    return await ProcesarDatosApi('POST', url, data);
}

export const ActualizarTipoInteraccion = async (data) => {
    const url = `${controlador}/actualizartipointeraccion`;
    return await ProcesarDatosApi('PUT', url, data);
}

export const InactivarTipoInteraccion = async (id) => {
    const url = `${controlador}/eliminartipointeraccion?idTipoInteraccion=${id}`;
    return await ProcesarDatosApi('DELETE', url);
}

export const ObtenerTipoInteraccionPorId = async (id) => {
    const url = `${controlador}/obtenertipointeraccionporid/${id}`;
    return await ProcesarDatosApi('GET', url);
}

export const ObtenerTipoInteraccion = async () => {
    const url = `${controlador}/obtenertipointeraccion`;
    return await ProcesarDatosApi('GET', url);
}
export const ObtenerTipoInteraccionActivos = async () => {
    const url = `${controlador}/obtenertipointeraccionactivos`;
    return await ProcesarDatosApi('GET', url);
}

File: BussinesLogic/Servicios/ServicioTipoInteraccion.cs
using DataAccess.StoredProcedures;
using Dominio.Dto;
using Dominio.Entiti;
using Dominio.Interfaces.Aplicacion.TipoInteraccion;
using Dominio.Interfaces.Infraestructura.BaseDatos;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace BussinesLogic.Servicios
{
    public class ServicioTipoInteraccion : IServicioTipoInteraccion
    {
        public readonly IRepositorioTipoInteraccion repositorioTipoInteraccion;

        public ServicioTipoInteraccion(IRepositorioTipoInteraccion repositorioTipoInteraccion)
        {
            this.repositorioTipoInteraccion = repositorioTipoInteraccion;
        }
        public async Task<DtoRespuestaSP> InsertarTipoInteraccion(EntitiTipoInteraccion entitiTipoInteraccion)
        {
            return await this.repositorioTipoInteraccion.InsertarTipoInteraccion(entitiTipoInteraccion);
        }
        public async Task<DtoRespuestaSP> ActualizarTipoInteraccion(EntitiTipoInteraccion entitiTipoInteraccion)
        {
            return await this.repositorioTipoInteraccion.ActualizarTipoInteraccion(entitiTipoInteraccion);
        }
        public async Task<DtoRespuestaSP> EliminarTipoInteraccion(int idTipoInteraccion)
        {
            return await this.repositorioTipoInteraccion.EliminarTipoInteraccion(idTipoInteraccion);
        }
        public async Task<DtoTipoInteraccion> ObtenerTipoInteraccionPorID(int idTipoInteraccion)
        {
            return await this.repositorioTipoInteraccion.ObtenerTipoInteraccionPorID(idTipoInteraccion);
        }

        public async Task<List<DtoTipoInteraccion>> ObtenerTipoInteraccion()
        {
            return await this.repositorioTipoInteraccion.ObtenerTipoInteraccion();
        }

        public async Task<List<DtoTipoInteraccion>> ObtenerTipoInteraccionActivos()
        {
            return await this.repositorioTipoInteraccion.ObtenerTipoInteraccionActivos();
        }
    }
}
